tests/unit/test_published_clusters.py

```diff
@@ -0,0 +1,53 @@
+import responses
+
+from tamr_unify_client import Client
+from tamr_unify_client.auth import UsernamePasswordAuth
+
+
+@responses.activate
+def test_published_clusters():
+
+    project_config_json = {
+        "id": "unify://unified-data/v1/projects/1",
+        "name": "Project_1",
+        "description": "Mastering Project",
+        "type": "DEDUP",
+        "unifiedDatasetName": "Project_1_unified_dataset",
+        "relativeId": "projects/1",
+        "externalId": "32b99cab-e01b-41e7-a29d-509165242c6f",
+    }
+
+    unified_dataset_json = {
+        "id": "unify://unified-data/v1/datasets/8",
+        "name": "Project_1_unified_dataset",
+        "version": "10",
+        "relativeId": "datasets/8",
+        "externalId": "Project_1_unified_dataset",
+    }
+
+    published_clusters_json = {
+        "id": "unify://unified-data/v1/datasets/32",
+        "name": "Project_1_unified_dataset_dedup_published_clusters",
+        "description": "All the mappings of records to clusters.",
+        "version": "253",
+        "relativeId": "datasets/32",
+        "externalId": "Project_1_unified_dataset_dedup_published_clusters",
+    }
+
+    datasets_json = [published_clusters_json]
+
+    unify = Client(UsernamePasswordAuth("username", "password"))
+    project_id = "1"
+
+    project_url = f"http://localhost:9100/api/versioned/v1/projects/{project_id}"
+    unified_dataset_url = (
+        f"http://localhost:9100/api/versioned/v1/projects/{project_id}/unifiedDataset"
+    )
+    datasets_url = f"http://localhost:9100/api/versioned/v1/datasets"
+
+    responses.add(responses.GET, project_url, json=project_config_json)
+    responses.add(responses.GET, unified_dataset_url, json=unified_dataset_json)
+    responses.add(responses.GET, datasets_url, json=datasets_json)
+    project = unify.projects.by_resource_id(project_id)
+    actual_published_clusters_dataset = project.as_mastering().published_clusters()
```

Comment:
After this, we need to call:
```
actual_published_clusters_dataset.refresh()
```
and make sure that goes to the right URL, `/api/versioned/v1/projects/1/publishedClusters:refresh`